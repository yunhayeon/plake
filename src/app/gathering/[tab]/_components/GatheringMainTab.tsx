import Link from "next/link";

import MainTab from "@/components/navigations/MainTab";
import { ONLINE, ONLINE_PATH } from "@/constants/gatheringFilterParams";
import { MAIN_TAB } from "@/constants/ui";
import useCustomSearchParams from "@/hooks/useCustomSearchParams";

interface IMainTabProps {
  pathname: string;
}

const GatheringMainTab = ({ pathname }: IMainTabProps) => {
  const { setSearchParams } = useCustomSearchParams();

  return (
    <div className="flex items-center gap-3">
      {MAIN_TAB.map((tab, i) => (
        <Link
          key={i}
          href={tab.href}
          aria-label={tab.name === "오프라인" ? "오프라인 탭" : "온라인 탭"}
          className="relative"
          onClick={() => pathname === ONLINE_PATH && setSearchParams(ONLINE)}
        >
          <MainTab active={pathname === tab.href} name={tab.name} />
        </Link>
      ))}
    </div>
  );
};

export default GatheringMainTab;
