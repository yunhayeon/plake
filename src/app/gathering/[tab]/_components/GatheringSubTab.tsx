import Link from "next/link";

import { Button } from "@/components/ui/Button";
import { OFFLINE_PATH, ONLINE_PATH } from "@/constants/gatheringFilterParams";
import { SUB_TAB } from "@/constants/ui";
import useCustomSearchParams from "@/hooks/useCustomSearchParams";
import { cn } from "@/lib/utils";
import useTabStore from "@/stores/useTabStore";

interface ISubTabProps {
  pathname: string;
}

const GatheringSubTab = ({ pathname }: ISubTabProps) => {
  const onSubTabChangeOn = useTabStore(state => state.onSubTabChangeOn);

  const { searchParams, setSearchParams } = useCustomSearchParams();
  const type = searchParams.get("type") || "";

  const href = pathname === OFFLINE_PATH ? OFFLINE_PATH : ONLINE_PATH;
  const isOffline = pathname === OFFLINE_PATH;

  return (
    <div className="flex items-center gap-2">
      {SUB_TAB[isOffline ? "OFFLINE" : "ONLINE"].map((tab, i) => (
        <Link
          href={tab.visibleValue ? `?type=${tab.visibleValue}` : href}
          key={i}
        >
          <Button
            key={i}
            variant="default"
            className={cn(
              "rounded-xl px-4 py-2.5",
              tab.visibleValue !== type &&
                "bg-gray-200 text-black hover:bg-gray-200/90",
            )}
            aria-label={
              isOffline ? `오프라인 ${tab.name} 탭` : `온라인 ${tab.name} 탭`
            }
            onClick={() => {
              setSearchParams({ type: tab.visibleValue });
              onSubTabChangeOn();
            }}
          >
            {tab.name}
          </Button>
        </Link>
      ))}
    </div>
  );
};

export default GatheringSubTab;
