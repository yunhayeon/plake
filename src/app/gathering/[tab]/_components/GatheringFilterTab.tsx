"use client";

import { usePathname } from "next/navigation";

import GatheringMainTab from "../_components/GatheringMainTab";
import GatheringSubTab from "../_components/GatheringSubTab";

const FilterTab = () => {
  const pathname = usePathname();

  return (
    <section className="flex flex-col gap-6" aria-label="주제 탭">
      <GatheringMainTab pathname={pathname} />
      <GatheringSubTab pathname={pathname} />
    </section>
  );
};

export default FilterTab;
