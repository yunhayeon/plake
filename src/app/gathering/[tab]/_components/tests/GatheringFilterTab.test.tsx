import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import GatheringFilterTab from "../GatheringFilterTab";
import GatheringMainTab from "../GatheringMainTab";
import GatheringSubTab from "../GatheringSubTab";

describe("FilterTab 컴포넌트 테스트", () => {
  it("GatheringMainTab 렌더링", () => {
    render(<GatheringMainTab pathname="/gathering/offline" />);

    expect(screen.getByLabelText("오프라인 탭")).toMatchSnapshot();
  });

  it("GatheringSubTab 렌더링", () => {
    render(<GatheringSubTab pathname="/gathering/offline" />);

    expect(screen.getByLabelText("오프라인 전체 탭")).toMatchSnapshot();
  });

  it("FilterTab 렌더링", async () => {
    render(<GatheringFilterTab />);

    expect(screen.getByLabelText("주제 탭")).toMatchSnapshot();
  });
});
