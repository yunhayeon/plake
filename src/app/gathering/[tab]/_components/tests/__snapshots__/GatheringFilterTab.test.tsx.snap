// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTab 컴포넌트 테스트 FilterTab 렌더링 1`] = `
<section
  aria-label="주제 탭"
  class="flex flex-col gap-6"
>
  <div
    class="flex items-center gap-3"
  >
    <a
      aria-label="오프라인 탭"
      class="relative"
      href="/gathering/offline"
    >
      <span
        class="relative pb-1.5 text-lg font-semibold text-gray-400"
      >
        오프라인
      </span>
    </a>
    <a
      aria-label="온라인 탭"
      class="relative"
      href="/gathering/online"
    >
      <span
        class="relative pb-1.5 text-lg font-semibold text-gray-400"
      >
        온라인
      </span>
    </a>
  </div>
  <div
    class="flex items-center gap-2"
  >
    <a
      href="/gathering/online"
    >
      <button
        aria-label="온라인 전체 탭"
        class="flex items-center justify-center whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 rounded-xl px-4 py-2.5"
        role="button"
      >
        전체
      </button>
    </a>
  </div>
</section>
`;

exports[`FilterTab 컴포넌트 테스트 GatheringMainTab 렌더링 1`] = `
<a
  aria-label="오프라인 탭"
  class="relative"
  href="/gathering/offline"
>
  <span
    class="relative pb-1.5 text-lg font-semibold text-black after:absolute after:mt-1.5 after:block after:w-full after:border-b-2 after:border-gray-900 after:content-['']"
  >
    오프라인
  </span>
</a>
`;

exports[`FilterTab 컴포넌트 테스트 GatheringSubTab 렌더링 1`] = `
<button
  aria-label="오프라인 전체 탭"
  class="flex items-center justify-center whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 rounded-xl px-4 py-2.5"
  role="button"
>
  전체
</button>
`;
